src/shared/cloudformation/templateRegistry.ts

```diff
@@ -0,0 +1,143 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path'
+import { fileURLToPath, pathToFileURL } from 'url'
+import * as vscode from 'vscode'
+import { FileWatcherListener } from '../utilities/fileSystemWatcher'
+import { CloudFormation } from './cloudformation'
+
+let templateRegistry: CloudFormationTemplateRegistry | undefined
+
+export function getTemplateRegistry(): CloudFormationTemplateRegistry {
+    if (!templateRegistry) {
+        throw new Error(
+            'Template Registry not initialized. Extension code should call activate() from shared/cloudformation/activation, test code should call setTemplateRegistry().'
+        )
+    }
+
+    return templateRegistry
+}
+
+export function setTemplateRegistry(registry: CloudFormationTemplateRegistry | undefined): void {
+    templateRegistry = registry
+}
```

Comment:
This can be made into a singleton instance, which instantiates on first access. First access will take place during activation, and will get initialized appropriately. See `CredentialsProviderManager.getInstance()` usage for an example.